protocol/x/clob/memclob/memclob.go

```diff
@@ -876,6 +889,20 @@ func (m *MemClobPriceTimePriority) matchOrder(
 		)
 		offchainUpdates.Append(matchOffchainUpdates)
 		writeCache()
+	} else {
+		// If state was not written to, re-send grpc stream updates for all orders
+		// involved in the match to "reset" fill amounts.
+		allUpdates := types.NewOffchainUpdates()
+		if !order.IsLiquidation() {
+			normalOrder := order.MustGetOrder()
+			updates := m.GetOrderbookUpdatesForOrderUpdate(ctx, normalOrder.OrderId)
+			allUpdates.Append(updates)
+		}
+		for _, fill := range newMakerFills {
+			updates := m.GetOrderbookUpdatesForOrderUpdate(ctx, fill.MakerOrderId)
+			allUpdates.Append(updates)
+		}
+		m.clobKeeper.SendOrderbookUpdates(ctx, allUpdates, false)
```

Comment:
Removing this didn't trigger any errors, but i think we still should have this -- It makes sense that we need to reset fill amounts if state was discarded.

2 qns
- do we need to reset anything else other than the order ids in the fills?
- do we need to reset orderbook placements for your side of streaming?

Reply:
1. I don't think so? think we only persist fills, right? other info (rewards, fees, etc) are irrelevent
2. no - because orderbook updates are persisted and not reverted even with failed placement